index.js

```diff
@@ -0,0 +1,242 @@
+                        let discordUser = await phoenix.User.getDiscordMapping(mapping.user.uuid);
+                        if (discordUser != null && discordUser.discord_id != null) {
+                            let roleVar = guild.roles.cache.find(role => role.name === "Crew");
+                            let member = await guild.members.fetch(discordUser.discord_id);
+                            await member.roles.remove(roleVar);
+                        }
+                    })
+                }
+            });
+        });
+    } catch (error) {
+        console.error("an error occured while deleting roles:", error);
+    }
+
+}
+
+async function updateRoles() {
+    //change this
+    const guild = phoenixClient.guilds.cache.get(phoenixGuildId);
+    let crews = await Promise.all((await phoenix.Crew.getCrews()));
+    crews.forEach(async (crew) => {
+        const allCrews = await phoenix.Crew.getCrew(crew.uuid);
+        const readableCrew = allCrews.positions;
+        readableCrew.forEach(async (position) => {
+            if (position.chief === true) {
+                position.position_mappings.forEach(async (mapping) => {
+                    try {
+                        let discordUser = await phoenix.User.getDiscordMapping(mapping.user.uuid);
+                        if (discordUser != null && discordUser.discord_id != null) {
+                            let roleVar = guild.roles.cache.find(role => role.name === "Gruppeleder");
+                            let member = await guild.members.fetch(discordUser.discord_id);
+                            await member.roles.add(roleVar);
+                        }
+                    } catch (error) {
+                        console.error("an error occured while editing roles:", error);
+                        console.log()
+                    }
+                })
+            } else {
+                position.position_mappings.forEach(async (mapping) => {
+                    try {
+                        let discordUser = await phoenix.User.getDiscordMapping(mapping.user.uuid);
+                        if (discordUser != null && discordUser.discord_id != null) {
+                            let roleVar = guild.roles.cache.find(role => role.name === "Crew");
+                            let member = await guild.members.fetch(discordUser.discord_id);
+                            await member.roles.add(roleVar);
+                        }
+                    } catch (error) {
+                        console.error("an error occured while editing roles:", error);
+                    }
+                })
+            }
+        });
+    });
+};
+
+
+//when the bot turns on
+phoenixClient.on("ready", () => {
+    console.log(`Logged in as ${phoenixClient.user.tag}!`);
+    updateRoles();
+    setInterval(function () {
+        updateRoles();
+        console.log("half an hour has gone by, updating roles");
```

Comment:
Every half hour? no rabbitmq?

Reply:
You specifically requested an additional check every half hour.